xbmc/messaging/ApplicationMessenger.cpp

```diff
@@ -241,13 +242,24 @@ void CApplicationMessenger::ProcessMessage(ThreadMessage *pMsg)
 
   CSingleLock lock(m_critSection);
   int mask = pMsg->dwMessage & TMSG_MASK_MESSAGE;
+  IMessageTarget* target;
+
+  try
+  {
+    target = m_mapTargets.at(mask);
+  }
+  catch (const std::out_of_range&)
+  {
+    target = nullptr;
+  }
 
-  auto target = m_mapTargets.at(mask);
```

Comment:
Maybe to use:
```cpp
  const auto it = m_mapTargets.find(mask);
  if (it != m_mapTargets.end() && it->second != nullptr)
  {
    CSingleExit exit(m_critSection);
    it->second->OnApplicationMessage(pMsg);
  }
  else
    CLog::LogF(LOGERROR, "receiver {} is not defined", mask);

```
Then a try/catch becomes not needed.

Reply:
Thanks. The nullptr check can be dripped in this case.

Updated.